assets/prototype/domain/shared/predicates/prices/updatePredicates.ts

```diff
@@ -0,0 +1,65 @@
+import { assoc, map, pickBy, when } from 'ramda';
+
+import { parsedAmount } from '../../../../application/utilities/money';
+import { Price } from '../../../eventEditor/data/types';
+import { isBasePrice, isPriceField } from './selectionPredicates';
+import { entityHasGuid } from '../shared/selectionPredicates';
+
+type updatePriceArrayProps = {
+	amount: number;
+	guid?: string;
+	prices?: Price[];
+	type?: string;
+};
+
+/**
+ * returns object with properties that match those of a Price entity
+ *
+ * @param {Price} price
+ */
+export const copyPriceFields = (price: Price): Price => pickBy<Price, Price>(isPriceField, price);
+
+/**
+ * updates the price amount
+ *
+ * @param {number} amount
+ */
+export const updatePriceAmount = (amount: number) => (price: Price): Price =>
+	assoc('amount', parsedAmount(amount || 0), price);
+
+/**
+ * updates the price type
+ *
+ * @param {string} type
+ */
+export const updatePriceType = (type: string) => (price: Price): Price => assoc('priceType', type, price);
```

Comment:
```suggestion
export const updatePriceType = (type: string) => (price: Price): Price => assoc<string, Price, string>('priceType', type, price);
```

Reply:
HA!
I had added that but then removed it because it appeared that TS was correctly inferring the types.
I have no problem adding those back in though, as it's better imho to be thorough with this "type" of thing ;)